### PR TITLE
IMPORT INTO delimited data support

### LINKED ISSUE #8290: importccl: support IMPORT INTO for DELIMITED
PR: https://github.com/cockroachdb/cockroach/pull/52628

From release notes:
> [`IMPORT INTO`](../v20.2/import-into.html) is now supported for `DELIMITED` and `PGCOPY` file formats. [#52628][#52628] {% comment %}doc{% endcomment %}

### PR DESCRIPTION
**Pull request by gemma-shay**
Added delimited data file info, import options, and examples. Restructured CSV & Avro import options to be a single table (similar to `IMPORT` page).

SQL Diagram still needs to be updated
Options table and example were mainly inferred from `IMPORT` and need to be fact-checked.

Closes #8290

### CODE REVIEW

**Review by lnhsingh — COMMENTED** [review]
:lgtm: , after you fix a couple of nits. lmk if you want me to re-review after the eng review

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/9966#-:-MVwqeDuEjBrQuXTGX6D:b-sfqyyn)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @gemma-shay)

___
*[_includes/v20.2/sql/diagrams/import_into.html, line 28 at r3](https://reviewable.io/reviews/cockroachdb/docs/9966#-MVwkQ1hA8LHuWu6qzbQ:-MVwkQ1hA8LHuWu6qzbR:b4u69yk) ([raw file](https://github.com/cockroachdb/docs/blob/531d78246f46cabac9de0c0934ea045726abad7a/_includes/v20.2/sql/diagrams/import_into.html#L28)):*
> ```HTML
> <rect x="108" y="249" width="52" height="32" rx="10"></rect>
> <rect x="106" y="247" width="52" height="32" class="terminal" rx="10"></rect>
> <text class="terminal" x="116" y="267">Avro</text>
> ```

nit: `Avro` should be `AVRO`

(applies to other version too)
___
*[_includes/v20.2/sql/diagrams/import_into.html, line 31 at r3](https://reviewable.io/reviews/cockroachdb/docs/9966#-MVwkSr-4H2xC4D0k0dK:-MVwkSr-4H2xC4D0k0dL:bq2jwv7) ([raw file](https://github.com/cockroachdb/docs/blob/531d78246f46cabac9de0c0934ea045726abad7a/_includes/v20.2/sql/diagrams/import_into.html#L31)):*
> ```HTML
> <rect x="108" y="293" width="84" height="32" rx="10"></rect>
> <rect x="106" y="291" width="84" height="32" class="terminal" rx="10"></rect>
> <text class="terminal" x="116" y="311">Delimited</text>
> ```

same: `Delimited` should be `DELIMITED`

(applies to other version too)
___
*[v20.2/import-into.md, line 58 at r3](https://reviewable.io/reviews/cockroachdb/docs/9966#-MVwkpVm-8v5htj8Snw_:-MVwkpVn5Ht0mnBxn9Y_:bo738ev) ([raw file](https://github.com/cockroachdb/docs/blob/531d78246f46cabac9de0c0934ea045726abad7a/v20.2/import-into.md#L58)):*
> ```Markdown
> #### Delimited data files
> 
> The `DELIMITED DATA` format can be used to import delimited data from any text file type, while ignoring characters that need to be escaped, like the following:
> ```

I think you should add the `New in 20.2` flag here
___
*[v20.2/import-into.md, line 71 at r3](https://reviewable.io/reviews/cockroachdb/docs/9966#-MVwpQibEBd0AC_tHGTg:-MVwpQibEBd0AC_tHGTh:bjmk9vs) ([raw file](https://github.com/cockroachdb/docs/blob/531d78246f46cabac9de0c0934ea045726abad7a/v20.2/import-into.md#L71)):*
> ```Markdown
> You can control the `IMPORT` process's behavior using any of the following key-value pairs as a `<option>  [= <value>]`.
> 
> <a name="delimiter"></a>
> ```

What's this anchor link for?
___
*[v20.2/import-into.md, line 73 at r3](https://reviewable.io/reviews/cockroachdb/docs/9966#-MVwphsF-WLbAo_fj4oM:-MVwphsF-WLbAo_fj4oN:b13nkvv) ([raw file](https://github.com/cockroachdb/docs/blob/531d78246f46cabac9de0c0934ea045726abad7a/v20.2/import-into.md#L73)):*
> ```Markdown
> <a name="delimiter"></a>
> 
> Key                 | <div style="width:130px">Context</div> | Value                                                                                                                             
> ```

This table LGTM from a grammar / style perspective, but make sure the eng / PM signs off on the accuracy of the DELIMITED options

**Review by gemma-shay — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/9966)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @gemma-shay and @lnhsingh)

___
*[_includes/v20.2/sql/diagrams/import_into.html, line 31 at r3](https://reviewable.io/reviews/cockroachdb/docs/9966#-MVwkSr-4H2xC4D0k0dK:-MVwtfEHBpkKAICD2Yv4:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/531d78246f46cabac9de0c0934ea045726abad7a/_includes/v20.2/sql/diagrams/import_into.html#L31)):*
<details><summary><i>Previously, lnhsingh (Lauren Hirata Singh) wrote…</i></summary><blockquote>

same: `Delimited` should be `DELIMITED`

(applies to other version too)
</blockquote></details>

Done.
___
*[v20.2/import-into.md, line 58 at r3](https://reviewable.io/reviews/cockroachdb/docs/9966#-MVwkpVm-8v5htj8Snw_:-MVwtg3h0qc1G15WnS81:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/531d78246f46cabac9de0c0934ea045726abad7a/v20.2/import-into.md#L58)):*
<details><summary><i>Previously, lnhsingh (Lauren Hirata Singh) wrote…</i></summary><blockquote>

I think you should add the `New in 20.2` flag here
</blockquote></details>

Done.
___
*[v20.2/import-into.md, line 71 at r3](https://reviewable.io/reviews/cockroachdb/docs/9966#-MVwpQibEBd0AC_tHGTg:-MVwtgYB27TpPTRTqIVX:b-hz3tig) ([raw file](https://github.com/cockroachdb/docs/blob/531d78246f46cabac9de0c0934ea045726abad7a/v20.2/import-into.md#L71)):*
<details><summary><i>Previously, lnhsingh (Lauren Hirata Singh) wrote…</i></summary><blockquote>

What's this anchor link for?
</blockquote></details>

Not sure, I think it was accidentally copied. Deleted.

File: v21.1/import-into.md [hunk 1]
```diff
@@ -266,6 +276,50 @@ Google Cloud:
...
+{% include copy-clipboard.html %}
+~~~ sql
+> IMPORT INTO customers
+    DELIMITED DATA (
+      's3://your-external-storage/employees.csv?AWS_ACCESS_KEY_ID=[placeholder]&AWS_SECRET_ACCESS_KEY=[placeholder]'
+    )
+    WITH
+      fields_terminated_by='|',
+      fields_enclosed_by='"',
+      fields_escaped_by='"';
```
Comment (pbardea) [inline v21.1/import-into.md:292]: nit: perhaps `\` may be a more typical value for `fields_escaped_by` and may be a better candidate in the example here?

**Review by pbardea — APPROVED** [review]
LGTM with a minor nit

**Pull request merged**